packages/util/test/bytes.spec.ts

```diff
@@ -351,3 +352,46 @@ tape('intToHex', function (st) {
   st.ok(intToHex(1) == '0x1', 'correctly converts 1 to a hex string')
   st.end()
 })
+
+tape('validateNoLeadingZeroes', function (st) {
+  const noLeadingZeroes = {
+    a: toBuffer('0x123'),
+  }
+  const leadingZeroes = {
+    a: toBuffer('0x001'),
+  }
+  const onlyZeroes = {
+    a: toBuffer('0x0'),
+  }
+  const emptyBuffer = {
+    a: toBuffer('0x'),
+  }
```

Comment:
Sorry, I am getting unsure here again, so just for a final confirmation so that this is not slipping through:

What is with a case like `0x01` (so the full byte just filled)? Manually tested on the current implementation of `validateNoLeadingZeroes` and this *passes* (since `b[0] === 1`).

Is this confirmed to be the correct behavior?

This would actually also be a good test case, just if someone has the occasion to add.